Sources/Instrumentation/Tracing/Span.swift

```diff
@@ -48,14 +48,19 @@ public protocol Span {
     /// - Parameter event: The `SpanEvent` to add to this `Span`.
     mutating func addEvent(_ event: SpanEvent)
 
-    // TODO: Wrap in a struct to hide collection implementation details.
-
     /// The attributes describing this `Span`.
     var attributes: SpanAttributes { get set }
 
     /// Returns true if this `Span` is recording information like events, attributes, status, etc.
     var isRecording: Bool { get }
 
+    /// The read-only collection of linked `Span`s.
+    var links: [SpanLink] { get }
+
+    /// Add a `SpanLink` in place.
+    /// - Parameter link: The `SpanLink` to add to this `Span`.
+    mutating func addLink(_ link: SpanLink)
```

Comment:
makes sense, would we though want to add extensions for just linking with a baggage context?

```
extension Span { 

  mutating func addLink(_ context: BaggageContext, attributes: SpanAttributes = [:]) { 
    self.addLink(SpanLink(context: context, attributes: attributes))
  }
}
```

Just a nice UX touch :-)

Reply:
Do you have guidance/rules on when to add such extensions? Personally, I'm always very careful not to add too much public API, even if it's in this case just one method.

Reply:
Your gut reaction is a good one 👍 Adding sugar is always to be considered if it's worth it... I'm ok to delay adding :) We can add if it gets annoying 👍 

// Here I kind of felt it'd be nice since the usage site I kind of imagine to be:

```swift
let span = ...

other.onComplete { context in 
    span.addLink(context)
    // continue things
}
```

But I think you may be right... let's not imagine but see in action and then we add it :)

Reply:
Glad to hear that 😊 I also like how your example snippet looks like. Maybe instead we could drop the method that takes a link in favor of adding this method.

Reply:
> Maybe instead we could drop the method that takes a link in favor of adding this method.

That's likely a not so good direction; it's better to err on the "the type" rather than the sugar and "i'll make the type for you", it's more explicit and adding the sugar on top :)

Reply:
Okay, that's a good point 👍